src/jit/compiler.cpp

```diff
@@ -447,62 +447,33 @@ void Compiler::getStructGcPtrsFromOp(GenTreePtr op, BYTE *gcPtrsOut)
 }
 #endif // FEATURE_MULTIREG_ARGS
 
-//------------------------------------------------------------------------
-// argOrReturnTypeForStruct: Get the "primitive" type, if any, that is used to pass or return
-//                           values of the given struct type.
-//
-// Arguments:
-//    clsHnd    - the handle for the struct type
-//    forReturn - true if we asking for this in a GT_RETURN context
-//                false if we are asking for this in a parameter passing context
-//
-// Return Value:
-//    The primitive type used to pass or return the struct, if applicable, or
-//    TYP_UNKNOWN otherwise.
-//
-// Assumptions:
-//    The given class handle must be for a value type (struct).
-//
-// Notes:
-//    Most of the work is done by the method of the same name that takes the
-//    size of the struct.
 
-
-var_types    Compiler::argOrReturnTypeForStruct(CORINFO_CLASS_HANDLE clsHnd, bool forReturn)
-{
-    unsigned size = info.compCompHnd->getClassSize(clsHnd);
-    return argOrReturnTypeForStruct(size, clsHnd, forReturn);
-}
-
-//------------------------------------------------------------------------
-// argOrReturnTypeForStruct: Get the "primitive" type, if any, that is used to pass or return
-//                           values of the given struct type.
+//-----------------------------------------------------------------------------
+// getPrimitiveTypeForStruct: 
+//     Get the "primitive" type that is is used for a struct 
+//     of size 'structSize'. 
+//     We examine 'clsHnd' to check the GC layout of the struct and 
+//     return TYP_REF for structs that simply wrap an object.
+//     If the struct is a one element HF, we will return the 
+//     proper floating point type.
 //
 // Arguments:
-//    size      - the size of the struct type
-//    clsHnd    - the handle for the struct type
-//    forReturn - true if we asking for this in a GT_RETURN context
-//                false if we are asking for this in a parameter passing context
+//    clsHnd     - the handle for the struct type, used when may have 
+//                 an HFA or if we need the GC layout for an object ref.
+//    structSize - the size of the struct type, cannot be zero
```

Comment:
Is there a reason you swapped the order of the argument descriptions, relative to how they are passed?